diff --git a/modules/cctbx_project/cctbx/symmetry_search/boost_python/symmetry_search_ext.cpp b/modules/cctbx_project/cctbx/symmetry_search/boost_python/symmetry_search_ext.cpp
index fcf86e75b3..c517e8a897 100644
--- a/modules/cctbx_project/cctbx/symmetry_search/boost_python/symmetry_search_ext.cpp
+++ b/modules/cctbx_project/cctbx/symmetry_search/boost_python/symmetry_search_ext.cpp
@@ -32,7 +32,7 @@ namespace cctbx { namespace symmetry_search { namespace boost_python {
       class_<wt>(name, no_init)
         .def(init<sgtbx::space_group const &,
                   af::const_ref<miller::index<> > const &,
-                  miller::f_calc_map<real_type> &,
+                  miller::f_calc_map<real_type> const &,
                   vector_type const &,
                   bool>
              ((arg("space_group"), arg("indices"), arg("f_c"),
diff --git a/modules/cctbx_project/cctbx/symmetry_search/translation_refinement.h b/modules/cctbx_project/cctbx/symmetry_search/translation_refinement.h
index 071ea55275..4827d001c8 100644
--- a/modules/cctbx_project/cctbx/symmetry_search/translation_refinement.h
+++ b/modules/cctbx_project/cctbx/symmetry_search/translation_refinement.h
@@ -45,7 +45,7 @@ struct symmetrised_shifted_structure_factors
   symmetrised_shifted_structure_factors(sgtbx::space_group const &space_group,
                                         af::const_ref<miller::index<> > const
                                         &indices,
-                                        miller::f_calc_map<real_type> &f_c,
+                                        miller::f_calc_map<real_type> const &f_c,
                                         vector_type const &x,
                                         bool compute_gradient)
   {
@@ -56,7 +56,7 @@ struct symmetrised_shifted_structure_factors
 
     int n = indices.size();
 
-    f_x.reserve(n);
+    this->f_x.reserve(n);
 
     math::cos_sin_exact<real_type> exp_i_2pi;
     for (int k=0; k<indices.size(); ++k) {
diff --git a/modules/cctbx_project/scitbx/array_family/boost_python/shared_flat_conversions.h b/modules/cctbx_project/scitbx/array_family/boost_python/shared_flat_conversions.h
index 7323899047..195b934eb5 100644
--- a/modules/cctbx_project/scitbx/array_family/boost_python/shared_flat_conversions.h
+++ b/modules/cctbx_project/scitbx/array_family/boost_python/shared_flat_conversions.h
@@ -28,7 +28,7 @@ namespace scitbx { namespace af { namespace boost_python {
 
     static PyTypeObject const *get_pytype() {
       using namespace boost::python;
-      return converter::registered<source_t>::converters.to_python_target_type();
+      return converter::registered<target_t>::converters.to_python_target_type();
     }
 
     /// Register conversions
